keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,2235 @@
+CRI-O and containerd runtimes have support for RDT and blockio classes and they
+provide an bridge-gap user interface through special pod annotations. The goal
+is to get these types of resources first class citizens and properly supported
+in Kubernetes, providing visibility, a well-defined user interface, and
+permission controls.
+
+We can identify two types, container-level and pod-level QoS-class resources.
+Container-level resources enable QoS on per-container granularity, for example
+container-level cgroups in Linux or cache and memory bandwidth control
+technologies. Examples for pod-level QoS include e.g. pod-level cgroups or
+network QoS that cannot support per-container granularity.
+
+### Goals
+
+- Make it possible to request QoS-class resources
+  - Support RDT class assignment of containers. This is already supported by
+    the containerd and CRI-O runtime and part of the OCI runtime-spec
+  - Support blockio class assignment of containers.
+  - Support Pod-level (sandbox-level) QoS-class resources
+- Make the API to support updating QoS-class resource assignment of running containers
+- Make the extensions flexible, enabling simple addition of other QoS-class
+  resource types in the future.
+- Make QoS-class resources opqaue (as possible) to the CRI client
+- Discovery of the available QoS-class resources
+- Resource status/capacity
+- Access control ([future work](#future-work))
+
+### Non-Goals
+
+- Interface or mechanism for configuring the QoS-class resources (responsibility of
+  the container runtime).
+- Specifying available (QoS-class) resource types or their detailed behavior
+- API changes to support updating Pod-level (sandbox-level) QoS-class resource
+  assignment of running pods (will be subject to a separate KEP)
+
+## Implementation phases
+
+This proposal splits the full implementation of QoS-class resources into
+multiple phases, building functionality gradually. The goal is to
+make the discussions more focused and easier. We may also learn on the way,
+insights from earlier phases affecting design choices made in the later phases,
+hopefully resulting in a better overall end result. However, we also outline
+all the future steps to not lose the overall big picture.
+
+The figure below illustrates the design of implementation phase 1. The first
+implementation phase covers the communication between kubelet and the container
+runtime (i.e. CRI API) and Kubernetes API for NodeStatus and PodSpec and their
+respecive implementations (kubelet and kube-scheduler). Access control (e.g.
+resource quota) is left to future work. This KEP (the [Proposal](#proposal) and
+[Design Details](#design-details)) in its current form implements this first
+phase.
+
+![design](./design.svg)
+
+QoS-class resources are designed to be opaque to the CRI client in the sense
+that the container runtime takes care of configuration and control of the
+resources and the classes within.
```

Comment:
and what have if the runtime does not implement it?

how an user can know if setting this parameters are going to work?

Reply:
In short, then the runtime does not advertise to kubelet (and it's not available on the node), i.e. it does not exist on the node from the K8s pov. Available QoS resource types (and their classes) can be seen from node status. Comparable to node resources in that respect, I'd say.

Reply:
Yes

Reply:
Thanks for the clarification